tests/platform/test_service_custom_objects.py

```diff
@@ -87,3 +87,14 @@ def test_custom_object_query_by_container(old_client):
     result = old_client.custom_objects.query_by_container("unittest")
     assert len(result.results) == 2
     assert result.total == 2
+
+
+def test_delete_by_container_and_key(old_client):
+    custom_object = old_client.custom_objects.create_or_update(
+        models.CustomObjectDraft(container="unittest", key="test-object-1", value=1234)
+    )
+    assert custom_object.id
+    assert custom_object.key == "test-object-1"
+    assert old_client.custom_objects.delete_by_container_and_key(
+        container=custom_object.container, key=custom_object.key
+    )
```

Comment:
You're asserting the function call. I think what you want to do here is assert something about the response. 
e.g.
```suggestion
    response = old_client.custom_objects.delete_by_container_and_key(
        container=custom_object.container, key=custom_object.key
    )
    assert response == ALL CUSTOM OBJECTS EXCEPT THE ONE YOU DELETED
```

Reply:
If you want to make doubly sure all is going well, you could expand to test to also explicitly check for a 404 response if you try to delete something that isnt there. For example by running the same delete again after the first set of asserts